growthmate-common/build.gradle

```diff
@@ -6,3 +6,7 @@ dependencies {
     testFixturesImplementation 'org.springframework.boot:spring-boot-starter-data-jpa'
     testFixturesImplementation 'org.springframework.boot:spring-boot-starter-test'
 }
+
+springBoot {
+    mainClass = 'com.growup.growthmate.GrowthMateApiApplication'
+}
```

Comment:
이것도 반드시 필요한 코드던가요? 이렇게 되면 common 모듈에서 api 모듈에 있는 클래스를 의존하게 되는데 이러면 양방향 의존이 되는 것이 아닐까요? api 모듈에서만 mainClass를 지정하고 common, core 모듈에 있는 gradle에는 mainClass 지정을 안하는 상태에선 스크립트가 실패할까요?

Reply:
- > Failed to calculate the value of task ':growthmate-common:bootJar' property 'mainClass'.
-       > Main class name has not been configured and it could not be resolved from classpath build/classes/java/main

Reply:
저도 그렇게 하고 싶었는데 위와 같이 각각 모듈에 넣어주지 않으면 mainClass 를 못 찾겠다고 뜨네요...ㅠ 저번 플젝 경험때도 이런 경험이 있었는데 결국 어떤 문제인지 찾지 못했습니다. 무엇이 문제일까요..?

Reply:
메인 클래스를 아무것도 지정하지 않은 상태에서 api 모듈만 빌드하면 빌드가 로컬에선 성공하긴 합니다. 이렇게 배포가 가능할까요?
`./gradlew growthmate-api:build`

저는 develop 브랜치에서 해본거긴합니다.

Reply:
제가 dockerfile을 api module jar 파일로 적어놔서 배포는 가능할것같은데 애초에 제가 develop 브랜치에서 빌드 했을떄 실패합니다. 

**Unsupported Java. 
Your build is currently configured to use Java 17.0.6 and Gradle 7.0.

Possible solution:
 - Use Java 16 as Gradle JVM: Open Gradle settings 
 - Upgrade Gradle wrapper to 7.2 version and re-import the project**
 
 이렇게 에러가 뜨고있네요...ㅠ